File: koweg-poc-payments/koweg-poc-payments-service/src/main/java/com/koweg/poc/payments/service/Audit.java
package com.koweg.poc.payments.service;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Audit {
    private final String auditContext;
    private final LocalDateTime date;
    private final String requestId;
    private final String correlationId;
    private final  String additional;

    public Audit(String auditContext, LocalDateTime date, String requestId, String correlationId, String additional){
        this.auditContext = auditContext;
        this.date = date;
        this.requestId = requestId;
        this.correlationId = correlationId;
        this.additional = additional;
    }

}

File: koweg-poc-audit/koweg-poc-audit-service/src/main/kotlin/com/koweg/poc/audit/Audit.kt
package com.koweg.poc.audit

import lombok.Data
import org.apache.commons.lang.builder.ReflectionToStringBuilder
import org.apache.commons.lang.builder.ToStringStyle
import java.time.LocalDateTime
@Data
class Audit (val auditContext: String, val date: LocalDateTime, val requestId: String, val correlationId: String ){
    override fun toString(): String = ReflectionToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE)
}